### ISSUE TITLE
Selecton preview box

### ISSUE THREAD

**Issue by Clemens-Ratte-Polle** [emvaized/selecton-extension#57]
Selecton also should have the ability to make a search in a layer.
So nobody has to open always a new tab.
It would raise productivity :)
You could make it like Link Previewer with a previewer box
https://chrome.google.com/webstore/detail/link-previewer/hhnejjlimgamhpolpjnjdkdcoclldhli

Best with a keyboard shortcut like alt+s
thx

**Comment by emvaized** [comment]
Hi there! 

I like this idea, but implementation becomes problematic. I worked with iframe popups for link preview before, and in my opinion they are quite unreliable and doesn't work with many websites due to securiy concerns. 

Also the UX could become quite messy – for example, if we reveal the popup on hovering the search button, it will replace the current popup for another search options. If we reveal it next to another sewrch option on hover – the whole interface will become quite cluttered and messy, and also what about the default search button? 

I better like the idea of opening the search page in new unmaximized window, for example when you shift+click button – this way all these issues could be solved, wgile still allowing you to search for text without opening the new tab. But here I don't like that this new window will have all browser controls, such as addressbar and tab bar. And if we will tell the window to hide them (by setting the window type "popup"), when there's no option to copy it's url or reopen it as a new tab in the main origin window... 

So, lot of things to think about with this idea

**Comment by Clemens-Ratte-Polle** [comment]
Thanks a lot :)
I can imagine what you mean but i am not firm too much with that ;)

Here are inspirations of addons that use similar techniques them you probaly know:

spotlight action popup  (registration needed?)
https://chrome.google.com/webstore/detail/spotlight/khfahkmjbdcfknfnfckndoijcbpfnfgi

or the total screen popup thing by
Saka
https://chrome.google.com/webstore/detail/saka/nbdfpcokndmapcollfpjdpjlabnibjdi

-
I testet many addons like yours and here are some that perhaps could help you with other "popups":

Link Previewer
https://chrome.google.com/webstore/detail/link-previewer/hhnejjlimgamhpolpjnjdkdcoclldhli

TabFloater: Picture-in-Picture for any tab!
https://chrome.google.com/webstore/detail/tabfloater-picture-in-pic/iojgbjjdoanmhcmmihbapiejfbbadhjd
-

For copying the URL of a popup there are many addons.
Copy link address 
https://chrome.google.com/webstore/detail/copy-link-address/kdejdkdjdoabfihpcjmgjebcpfbhepmh?hl=de

-
OK, i do not want to mess up with links so i hope that you keep on going :)
You should also show a link for spending "a cup of coffe" :D

**Comment by emvaized** [comment]
Thanks for enhanced feedback. 
Yeah, the TabFloater idea looks the most similar to what I had on my mind. However, the problem of this idea still remains — imagine you opened such popup window with search, then clicked few links, changed few pages, and now you found something you want to save for further reading and get back to your main window. 

Your solution is installing another extension, which adds "Copy page url" entry in the context menu — but it doesn't sound very convenient (because you still have to close the popup, create new tab and paste the url), and it requires installing another extension. Sure, Selecton could do it on itself, adding such context menu entry — but it would be useful only in this specific use cases, and some users may find it unneccessary bloat... 

I would probably explore this idea a bit more, maybe create another extension solely for this purpose, and if I find a working formula maybe I'll include this as option in Selecton. But could not promise anything, as I don't actively use and develop Selecton now (probably should better mention that somewhere)

**Comment by Clemens-Ratte-Polle** [comment]
thank you :)

**Comment by Clemens-Ratte-Polle** [comment]
thy, perhaps also as Chrome Addon not only Firefox addon? :)

**Comment by emvaized** [comment]
> thy, perhaps also as Chrome Addon not only Firefox addon? :)

Here's version you can install in Chrome. But you'll have to install it manually on the Extensions page (it may require to enable Developer mode on the same page):
[0.0.6 (beta).zip](https://github.com/emvaized/selecton-extension/files/14470804/0.0.6.beta.zip)

**Comment by emvaized** [comment]
In the version `3.7.9` tooltip buttons were converted to behave like regular links (`<a>` html element). It is possible now to use any third-party extension to preview them or open in a popup window. I believe it to be much better solution than incorporating popup previewing logic in the Selecton itself. 

I also published my experimental extension under the name [Open In Popup Window](https://github.com/emvaized/open-in-popup-window-extension). With it installed and configured, you can quickly open any Selecton button in a popup window by using <kbd>Shift</kbd> + Click on it: 

<img width="918" alt="Screenshot" src="https://github.com/user-attachments/assets/482cda92-b204-4f98-9f5f-e0b9202d4b15">

**Comment by tranhungkcn** [comment]
Thanks for your help. I tested [Open In Popup Window](https://github.com/emvaized/open-in-popup-window-extension), combined with Selecton, and I can get the function as I hoped.
* But there is still 1 bug, Selecton I used the option Appearance>Buttons style: Icons only
Then when I shift click on the button which pop up by Selecton, if I click at the center of the button (with icon on it) the [Open In Popup Window] will understand the  icon as a picture and pop up a window with that "picture" of icon, instead of pop up a window of the search or translate link. The attached file is screen shoot when I shift click on button translate which pop up by [Selecton]
![screenshoot 2024-11-11 114009](https://github.com/user-attachments/assets/75de5493-33f1-46c2-b51b-8cbfc3b3df15)
One more problem, I tried to unselect the option "Enable option to view images in popup window" of  [Open In Popup Window], but this function still working, with both normal picture and the icon on pop up button of [Selecton]
